common/models/model-definition.js

```diff
@@ -173,7 +182,8 @@ function ready(ModelDefinition) {
     for (var ix in relatedData) {
       var data = Entity.getConfigFromData(relatedData[ix]);
       delete data[relation.foreignKey];
-      delete data[relation.embed.key];
+      delete data.name;
```

Comment:
I don't think we should always delete `data.name`, we should `delete data[relation.embed.key]` instead when the key is set. When it's not set and there is `keySetter`, then it's the responsibility of `getConfigFromData` to remove the `name` property if it is not expected in the output.

The idea is to keep `cleanRelatedData` totally not opinionated, just providing extension hooks allowing domain entities to implement whatever custom rules they need.